src/frontend/src/lib/derived/address.derived.ts

```diff
@@ -1,19 +1,24 @@
-import { addressStore } from '$lib/stores/address.store';
+import { ETHEREUM_TOKEN_ID } from '$env/tokens.env';
+import { addressStore, type AddressData } from '$lib/stores/address.store';
 import type { OptionEthAddress } from '$lib/types/address';
-import { isNullish } from '@dfinity/utils';
+import { isNullish, nonNullish } from '@dfinity/utils';
 import { derived, type Readable } from 'svelte/store';
 
 export const addressNotLoaded: Readable<boolean> = derived([addressStore], ([$addressStore]) =>
 	isNullish($addressStore)
 );
 
 export const ethAddress: Readable<OptionEthAddress> = derived([addressStore], ([$addressStore]) =>
-	$addressStore === null ? null : $addressStore?.address
+	$addressStore?.[ETHEREUM_TOKEN_ID] === null ? null : $addressStore?.[ETHEREUM_TOKEN_ID].data
```

Comment:
`$addressStore?.[ETHEREUM_TOKEN_ID].data` could not potentially through an error? i.e. `$addressStore?.[ETHEREUM_TOKEN_ID]` can potentially be `undefined`?

Reply:
indeed, but previously that was how it was treated too, meaning that when it was just `$addressStore` (referring to ETH address), it could have been `string | undefined | null`, and the derived was accepting undefined for the right-side of the condition too (as it is in our case now).

I didn't want to change the logic

Reply:
You did not understand my feedback: this `$addressStore?.[ETHEREUM_TOKEN_ID].data` can throw an error if `$addressStore?.[ETHEREUM_TOKEN_ID] === undefined`.

Reply:
i.e. => isn't optional chaining missing `$addressStore?.[ETHEREUM_TOKEN_ID]?.data` ?

Reply:
ah! yes! you are right, but then why was not raised by lint? strange... in the store right after it raised it

corrected